### PR TITLE
terminal-typeracer: init at 2.0.2

### PR DESCRIPTION
**Pull request by yoctocell**
###### Motivation for this change
Adds [terminal-typeracer](https://gitlab.com/ttyperacer/terminal-typeracer), a TUI for typeracer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by dasj19 — APPROVED** [review]
Built successfully with `nix-shell -p nixpkgs-review --run "nixpkgs-review pr 93697"` and the application runs successfully.
Approved.

Welcome to nix!

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8845,6 +8845,12 @@
     githubId = 1322201;
     name = "Yochai";
   };
+  yoctocell = {
```
Comment (sikmir) [inline maintainers/maintainer-list.nix:8848]: IIRC, adding yourself as a maintainer is preferred to be a separate commit.

Something like, `maintainers: add yoctocell`. Then apply the terminal-typeracer commit.

**Review by sikmir — APPROVED** [review]

**Review by robertodr — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 93697` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terminal-typeracer</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Builds on darwin with the following patch:
```patch
diff --git a/pkgs/applications/misc/terminal-typeracer/default.nix b/pkgs/applications/misc/terminal-typeracer/default.nix
index 46ee0807c621..88f5cfefca7c 100644
--- a/pkgs/applications/misc/terminal-typeracer/default.nix
+++ b/pkgs/applications/misc/terminal-typeracer/default.nix
@@ -4,6 +4,8 @@
 , pkg-config
 , openssl
 , sqlite
+, libiconv
+, Security
 }:

 rustPlatform.buildRustPackage rec {
@@ -19,7 +21,7 @@ rustPlatform.buildRustPackage rec {

   cargoSha256 = "0ky8m23fjjbv7fr9776fjczpg0d43jxwnjxjpwz56jpynwnihfkl";

-  buildInputs = [ openssl sqlite ];
+  buildInputs = [ openssl sqlite ] ++ stdenv.lib.optionals stdenv.isDarwin [ libiconv Security ];
   nativeBuildInputs = [ pkg-config ];

   meta = with stdenv.lib; {
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 5d35887d91cf..38b0013e09b9 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -24206,7 +24206,9 @@ in

   twinkle = qt5.callPackage ../applications/networking/instant-messengers/twinkle { };

-  terminal-typeracer = callPackage ../applications/misc/terminal-typeracer { };
+  terminal-typeracer = callPackage ../applications/misc/terminal-typeracer {
+    inherit (darwin.apple_sdk.frameworks) Security;
+  };

   ueberzug = with python3Packages; toPythonApplication ueberzug;
```

File: pkgs/applications/misc/terminal-typeracer/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = [ openssl sqlite ] ++ stdenv.lib.optionals stdenv.isDarwin [ libiconv Security ];
+  nativeBuildInputs = [ pkg-config ];
+
+  meta = with stdenv.lib; {
+    description = "An open source terminal based version of Typeracer written in rust";
+    homepage = "https://gitlab.com/ttyperacer/terminal-typeracer";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ yoctocell ];
+    platforms = platforms.x86_64;
```
Comment (andir) [inline pkgs/applications/misc/terminal-typeracer/default.nix:32]: Anything keeping us from setting this to `platforms.linux`?

Comment (andir) [reply pkgs/applications/misc/terminal-typeracer/default.nix:32]: Or even all now that darwin works?

Comment (yoctocell) [reply pkgs/applications/misc/terminal-typeracer/default.nix:32]: I am not sure if it works on aarch64.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 93697` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terminal-typeracer</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 93697` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terminal-typeracer</li>
  </ul>
</details>
